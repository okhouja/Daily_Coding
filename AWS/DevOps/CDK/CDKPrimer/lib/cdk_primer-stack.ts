// import { Construct } from 'constructs';
// // import * as sqs from 'aws-cdk-lib/aws-sqs';

// export class CdkPrimerStack extends cdk.Stack {
//   constructor(scope: Construct, id: string, props?: cdk.StackProps) {
//     super(scope, id, props);

//     // The code that defines your stack goes here

//     // example resource
//     // const queue = new sqs.Queue(this, 'CdkPrimerQueue', {
//     //   visibilityTimeout: cdk.Duration.seconds(300)
//     // });
//   }
// }

// import * as cdk from "@aws-cdk/core";
// import * as ec2 from "@aws-cdk/aws-ec2";
// import * as ecs from "@aws-cdk/aws-ecs";
// import * as ecs_patterns from "@aws-cdk/aws-ecs-patterns";


import * as cdk from 'aws-cdk-lib';

import {aws_ecs as ecs,aws_ec2 as ec2,aws_ecs_patterns as ecs_patterns}  from 'aws-cdk-lib';
import { Construct } from "constructs";

export class CdkPrimerStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const vpc = new ec2.Vpc(this, "MyVpc", { maxAzs: 2 });

    const cluster = new ecs.Cluster(this, "MyCluster", { vpc: vpc });

    new ecs_patterns.ApplicationLoadBalancedFargateService(
      this,
      "MyFargateService",
      {
        cluster: cluster,

        taskImageOptions: {
          image: ecs.ContainerImage.fromRegistry("amazon/amazon-ecs-sample"),
        },

        publicLoadBalancer: true,
      }
    );
  }
}

