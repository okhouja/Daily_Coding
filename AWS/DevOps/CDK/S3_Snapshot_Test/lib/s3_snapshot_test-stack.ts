import { Stack, StackProps } from 'aws-cdk-lib';
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { aws_s3 as s3 } from 'aws-cdk-lib';

export class S3SnapshotTestStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    new s3.Bucket(this, "snapshotdemobucket", {
      versioned: true,
      objectOwnership: s3.ObjectOwnership.BUCKET_OWNER_PREFERRED,
      encryption: s3.BucketEncryption.S3_MANAGED,
      lifecycleRules: [
        {
          expiration: cdk.Duration.days(25)
        }
      ],
      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
      removalPolicy: cdk.RemovalPolicy.DESTROY
    });
  }
}
