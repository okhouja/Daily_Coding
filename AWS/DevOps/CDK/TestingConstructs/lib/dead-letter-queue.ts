import * as cloudwatch from 'aws-cdk-lib/aws-cloudwatch'
import { Stack, StackProps,Duration } from 'aws-cdk-lib';

// An sqs queue for unsuccessful invocations of a lambda function
import * as sqs from 'aws-cdk-lib/aws-sqs'
import { Construct } from 'constructs';

export class DeadLetterQueue extends Stack {
  public readonly messagesInQueueAlarm: cloudwatch.IAlarm;

  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);


    // const metric = queue.metric("ApproximateNumberOfMessagesVisible");
    const metric = new cloudwatch.Metric({
        namespace: 'MyNamespace',
        metricName: 'MyMetric',
        dimensionsMap: { MyDimension: 'MyDimensionValue' },

        // add a period property of 6 minute to override the default of 5 minutes
        period: Duration.minutes(6)
      });

    // Add the alarm
    this.messagesInQueueAlarm = new cloudwatch.Alarm(this, 'Alarm', {
      alarmDescription: 'There are messages in the Dead Letter Queue',
      evaluationPeriods: 1,
      threshold: 1,
      metric: metric,
      
    });
  }
}