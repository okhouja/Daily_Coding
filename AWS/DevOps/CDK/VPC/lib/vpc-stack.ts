import * as cdk from "aws-cdk-lib";
// import {aws_ecs as ecs,aws_ec2 as ec2,aws_ecs_patterns as ecs_patterns}  from 'aws-cdk-lib';
// import { IpAddresses } from "aws-cdk-lib/aws-ec2";
import * as ec2 from "aws-cdk-lib/aws-ec2";

import { Construct } from "constructs";
// import * as sqs from 'aws-cdk-lib/aws-sqs';

export class VpcStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // new ec2.CfnVPC(this, 'MyVPC', { cidrBlock: '10.0.0.0/16'})

    // new ec2.Vpc(this, "MyVPC", { cidr: "10.0.0.0/16" });

    new ec2.Vpc(this, "MyVPC", {
      ipAddresses: ec2.IpAddresses.cidr("10.0.0.0/16"),
    });
  }
}
