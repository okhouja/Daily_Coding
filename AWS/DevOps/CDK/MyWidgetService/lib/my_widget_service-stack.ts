import * as cdk from '@aws-cdk/core';
// import * as cdk from 'aws-cdk-lib';
import * as widget_service from '../lib/widget_service';
import { Construct } from 'constructs';
// import * as sqs from 'aws-cdk-lib/aws-sqs';

export class MyWidgetServiceStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    new widget_service.WidgetService(this, 'Widgets');
  }
}
