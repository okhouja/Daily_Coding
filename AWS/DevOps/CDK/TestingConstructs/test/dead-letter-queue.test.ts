// import * as cdk from 'aws-cdk-lib';
// import '@aws-cdk/assert/jest';
import { Template } from 'aws-cdk-lib/assertions';
import {App } from 'aws-cdk-lib/core';

import {DeadLetterQueue} from '../lib/dead-letter-queue';

// test('dlq creates an alarm', () => {
//   const app = new App();
//   const dlqStack = new DeadLetterQueue(app, 'DLQ');
// const template = Template.fromStack(dlqStack)

//   // expect(template.toJSON()).toMatchSnapshot();;

//   expect(template.hasResourceProperties('AWS::CloudWatch::Alarm', {
//     MetricName: 'ApproximateNumberOfMessagesVisible',
//     Namespace: 'AWS/SQS',
// }))
// });