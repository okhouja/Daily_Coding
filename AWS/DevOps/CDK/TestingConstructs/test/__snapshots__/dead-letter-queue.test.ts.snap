// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dlq creates an alarm 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "Alarm7103F465": {
      "Properties": {
        "AlarmDescription": "There are messages in the Dead Letter Queue",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": [
          {
            "Name": "MyDimension",
            "Value": "MyDimensionValue",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "MyMetric",
        "Namespace": "MyNamespace",
        "Period": 360,
        "Statistic": "Average",
        "Threshold": 1,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
